crates/proto/src/native_tls/tls_client_stream.rs

```diff
@@ -57,6 +57,39 @@ impl<S: Connect> TlsClientStreamBuilder<S> {
         self.0.bind_addr(bind_addr);
     }
 
+    /// Creates a new TlsStream to the specified name_server with stream future.
+    ///
+    /// # Arguments
+    ///
+    /// * 'future` - future of TCP stream
+    /// * `name_server` - IP and Port for the remote DNS resolver
+    /// * `dns_name` - The DNS name, Subject Public Key Info (SPKI) name, as associated to a certificate
+    #[allow(clippy::type_complexity)]
+    pub fn build_with_future<F>(
+        self,
+        future: F,
+        name_server: SocketAddr,
+        dns_name: String,
+    ) -> (
```

Comment:
Sorry I don't quite get what you mean. Do you mean that this interface lack `client_config: Arc<ClientConfig>`? It's because this is native_tls and `ClientConfig` is for rustls only.

Reply:
Really just the name.

Reply:
It's fine.